pkg/bench/rttanalysis/validate_benchmark_data.go

```diff
@@ -58,193 +60,152 @@ var (
+			return
+		}
+		if !exp.matches(r.result) {
+			t.Errorf("fail: expected %s to perform KV lookups in [%d, %d], got %d",
+				r.name, exp.min, exp.max, r.result)
+		} else {
+			t.Logf("success: expected %s to perform KV lookups in [%d, %d], got %d",
+				r.name, exp.min, exp.max, r.result)
+		}
 	})
-	require.NoErrorf(t, cmd.Start(), "failed to start command %v", cmd)
-	require.NoError(t, g.Wait())
-	require.NoErrorf(t, cmd.Wait(), "failed to wait for command %v", cmd)
-	return readBenchmarkResults(t, &stdoutBuf)
 }
 
-var (
-	benchmarkResultRe = regexp.MustCompile(`^Benchmark(?P<testname>.*)-\d+\s.*\s(?P<roundtrips>\S+)\sroundtrips$`)
-	testNameIdx       = benchmarkResultRe.SubexpIndex("testname")
-	roundtripsIdx     = benchmarkResultRe.SubexpIndex("roundtrips")
-)
-
-func readBenchmarkResults(t *testing.T, benchmarkOutput io.Reader) []benchmarkResult {
-	var ret []benchmarkResult
-	sc := bufio.NewScanner(benchmarkOutput)
-	for sc.Scan() {
-		match := benchmarkResultRe.FindStringSubmatch(sc.Text())
-		if match == nil {
-			continue
+func mergeExpectations(existing, new benchmarkExpectations) (merged benchmarkExpectations) {
+	sort.Sort(existing)
+	sort.Sort(new)
+	pop := func(be *benchmarkExpectations) (ret benchmarkExpectation) {
+		ret = (*be)[0]
+		*be = (*be)[1:]
+		return ret
+	}
+	for len(existing) > 0 && len(new) > 0 {
+		switch {
+		case existing[0].name < new[0].name:
+			merged = append(merged, pop(&existing))
+		case existing[0].name > new[0].name:
+			merged = append(merged, pop(&new))
+		default:
+			pop(&existing) // discard the existing value if they are equal
+			merged = append(merged, pop(&new))
 		}
-		testName := match[testNameIdx]
-		got, err := strconv.ParseFloat(match[roundtripsIdx], 64)
-		require.NoError(t, err)
-		ret = append(ret, benchmarkResult{
-			name:   testName,
-			result: int(got),
-		})
 	}
-	require.NoError(t, sc.Err())
-	return ret
+	// Only one of existing or new will be non-empty.
+	merged = append(append(merged, new...), existing...)
+	return merged
 }
 
-// rewriteBenchmarkExpectations re-runs the specified benchmarks and throws out
-// the existing values in the results file. All other values are preserved.
-func rewriteBenchmarkExpectations(t *testing.T, benchmarks []string) {
-
-	// Split off the filter so as to avoid spinning off unnecessary subprocesses.
-	slashIdx := strings.Index(*rewriteFlag, "/")
-	var afterSlash string
-	if slashIdx == -1 {
-		slashIdx = len(*rewriteFlag)
-	} else {
-		afterSlash = (*rewriteFlag)[slashIdx+1:]
-	}
-	benchmarkFilter, err := regexp.Compile((*rewriteFlag)[:slashIdx])
-	require.NoError(t, err)
-
-	var g errgroup.Group
-	resChan := make(chan []benchmarkResult)
-	run := func(b string) {
-		if !benchmarkFilter.MatchString(b) {
+// execTestSubprocess execs the testing binary with all the same flags in order
+// to run it without metamorphic testing enabled. Metamorphic testing messes
+// with the benchmark results.
```

Comment:
so are we just running this to pass some tests? is that strictly necessary?
i'm confused why we don't just ignore metamorphic builds